core/src/agent/state.rs

```diff
@@ -144,10 +147,18 @@ pub enum ActionResponse {
     LinkEntries(Result<Entry, HolochainError>),
 }
 
-pub fn create_new_chain_header(entry: &Entry, agent_state: &AgentState) -> ChainHeader {
+pub fn create_new_chain_header(entry: &Entry, context: Arc<Context>) -> ChainHeader {
+    let agent_state = context
+        .state()
+        .expect("create_new_chain_header called without state")
+        .agent();
+    let agent_address = agent_state
+        .get_agent_address()
+        .unwrap_or(context.agent_id.address());
     ChainHeader::new(
         &entry.entry_type(),
         &entry.address(),
+        &vec![agent_address],
```

Comment:
This is the most important line in this PR ;)

Reply:
as in: everything else needed to happen as a consequence to get this line in :D

Reply:
![image](https://user-images.githubusercontent.com/629710/49497153-c4da1700-f8bb-11e8-8412-60017288b0d2.png)

Reply:
yup ;)